### PR TITLE
Minor fixes in AggregateFunctionSegmentLengthSumData

### PR DESCRIPTION
**Pull request by vdimir**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Fixes for https://github.com/ClickHouse/ClickHouse/pull/24250#issuecomment-846469120

### CODE REVIEW

File: src/AggregateFunctions/AggregateFunctionIntervalLengthSum.cpp [hunk 1]
```diff
@@ -32,35 +32,35 @@ namespace
 
         if (WhichDataType{args.begin()[0]}.idx != WhichDataType{args.begin()[1]}.idx)
             throw Exception(
-                "Illegal type " + args.begin()[0]->getName() + " and " + args.begin()[1]->getName() + " of arguments of aggregate function "
-                    + name + ", there two arguments should have same DataType",
+                "Illegal types " + args.begin()[0]->getName() + " and " + args.begin()[1]->getName() + " of arguments of aggregate function "
+                    + name + ", there two arguments should have same data types",
```
Comment (Avogar) [inline src/AggregateFunctions/AggregateFunctionIntervalLengthSum.cpp:36]:
```suggestion
                    + name + ", both arguments should have same data types",
```

File: src/AggregateFunctions/AggregateFunctionIntervalLengthSum.cpp [hunk 2]
```diff
@@ -32,35 +32,35 @@ namespace
...
                 ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 
         for (const auto & arg : args)
         {
             if (!isNativeNumber(arg) && !isDateOrDateTime(arg))
```
Comment (Avogar) [inline src/AggregateFunctions/AggregateFunctionIntervalLengthSum.cpp:41]: `isDateOrDateTime` also checks `DateTime64`, use `isDate` and `isDateTime`. We are going to remove `isDateOrDateTime` at all in https://github.com/ClickHouse/ClickHouse/pull/24398 (see resolved discussion).

**Review by Avogar — APPROVED** [review]

**Comment by Avogar** [comment]
@vdimir can you resolve conflicts? I guess it's after merging https://github.com/ClickHouse/ClickHouse/pull/24398

**Comment by vdimir** [comment]
> @vdimir can you resolve conflicts? I guess it's after merging #24398

Rebased on `master`

**Pull request merged**